applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -106,6 +119,66 @@ static int next_free_audio_stream_index_get(uint8_t *index)
 	return -ENOSPC;
 }
 
+/**
+ * @brief  Get the presentation delay for all headsets
+ *
+ * @param  mode   The mode of search for the presentation delay
+ * @param  index  The index of the hesdset to test against
+ *
+ * @return The chosen presentation delau in micro seconds or 0 if error
+ */
+static uint32_t find_headset_pres_delay(enum le_audio_pres_search mode, int index)
+{
+	uint32_t pres_dly_us = 0;
+	uint32_t pres_dly_min = headsets[index].pref_qos.pd_min;
+	uint32_t pres_dly_max = headsets[index].pref_qos.pd_max;
+	uint32_t pref_dly_min = headsets[index].pref_qos.pref_pd_min;
+	uint32_t pref_dly_max = headsets[index].pref_qos.pref_pd_max;
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		if (headsets[i].sink_stream != NULL) {
+			pres_dly_min = MAX(pres_dly_min, headsets[i].pref_qos.pd_min);
+			pres_dly_max = MIN(pres_dly_max, headsets[i].pref_qos.pd_max);
+			pref_dly_min = MAX(pref_dly_min, headsets[i].pref_qos.pref_pd_min);
+			pref_dly_max = MIN(pref_dly_max, headsets[i].pref_qos.pref_pd_max);
+		}
+	}
+
+	switch (mode) {
+	case AUDIO_PRES_DELAY_MODE_MIN:
+		pres_dly_us = pres_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_MAX:
+		pres_dly_us = pres_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MIN:
+		pres_dly_us = pref_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MAX:
+		pres_dly_us = pref_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_LOCAL:
+		if (pres_dly_min <= CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US &&
+		    pres_dly_max >= CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US) {
+			pres_dly_us = CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US;
+		} else {
+			pres_dly_us = 0;
```

Comment:
Should return an error with a negative value

Reply:
The return is the presentation delay and is a uint32_t so I will need to change the interface to either return int32_t and the return will be an error or a signed delay value. The later makes no sense and I use zero as it is not a valid delay, we have to have something! Is this the standard, as there are other functions that return uint32_t and use zero as an error value.

Reply:
I recommend to take pres_dly_us as a pointer to this function and return only errors. This is how it is done for the most part.

Reply:
Looking at it again, I think that if the local pres delay is outside of the max/min range of the headsets, then it should revert to the closest possible value, but print a warning. Meaning that if gateway wants to set pres delay to 10ms but the headsets support a range of 20-40ms, then 20ms should be set.

Reply:
Can we discuss?

Reply:
yeah, let's discuss it after the stand up if we have time.

Reply:
Updated to send closest minimum or maximum allowed over all audio receivers.